pkg/utils/kubernetes/kubernetes.go

```diff
@@ -190,27 +192,62 @@ func WaitUntilResourceDeletedWithDefaults(ctx context.Context, c client.Client,
 
 // WaitUntilLoadBalancerIsReady waits until the given external load balancer has
 // been created (i.e., its ingress information has been updated in the service status).
-func WaitUntilLoadBalancerIsReady(ctx context.Context, c client.Client, namespace, name string, timeout time.Duration, logger logrus.FieldLogger) (string, error) {
+func WaitUntilLoadBalancerIsReady(
+	ctx context.Context,
+	log interface{}, // TODO(rfranzke): Use logr.Logger when all usages are adapted
+	c client.Client,
+	namespace, name string,
+	timeout time.Duration,
+) (
+	string,
+	error,
+) {
 	var (
 		loadBalancerIngress string
 		service             = &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}}
 	)
 
+	var logger interface{}
+	switch l := log.(type) {
+	case logrus.FieldLogger:
+		logger = l.WithField("service", client.ObjectKeyFromObject(service))
+	case logr.Logger:
+		logger = l.WithValues("service", client.ObjectKeyFromObject(service))
+	default:
+		logger = logf.Log.WithValues("service", client.ObjectKeyFromObject(service))
```

Comment:
Shouldn't the default case be handled when adding log messages below as well?

Reply:
Not needed since by default this created a `logr.Logger` which is already handled :)